File: udp/servidor.py
import socket
import json

def start_server(host: str, port: int):
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    
    server_socket.bind((host, port))

    print(f'Servidor UDP iniciado em {host}:{port}')

    while True:
        data, adress = server_socket.recvfrom(1024)
        message = data.decode('utf-8')

        name = message.split("!=")[0]
        message = message.split("!=")[1]

        print(f'[{name.upper()}]: {message}') 

if __name__=='__main__':
    HOST = 'localhost'
    PORT = 9000

    start_server(HOST, PORT)

File: udpp/servidor.js
const dgram = require('dgram');
const server = dgram.createSocket('udp4');

const PORT = 9000;
const HOST = 'localhost';

server.on('listening', () => {
    console.log(`Servidor UDP iniciado em ${HOST}:${PORT}`);
});

server.on('message', (msg, rinfo) => {
    const message = msg.toString();
    const [name, content] = message.split('!=');
    console.log(`[${name.toUpperCase()}]: ${content}`);
});

server.bind(PORT, HOST);
